roborock/api.py

```diff
@@ -113,6 +113,13 @@ def _async_response(
         self, request_id: int, protocol_id: int = 0
     ) -> Coroutine[Any, Any, tuple[Any, VacuumError | None]]:
         queue = RoborockFuture(protocol_id)
+        if request_id in self._waiting_queue:
+            new_id = get_next_int(10000, 32767)
+            _LOGGER.warning(
+                f"Attempting to create a future with an existing request_id... New id is {new_id}. "
+                f"Code may not function properly."
+            )
+            request_id = new_id
```

Comment:
I think this ends up writing the future with a different key, but does not change the request id chosen by the caller. I think this means the response will still be left hanging and/or associated with the wrong request?